ILSpy/SearchPane.cs

```diff
@@ -242,10 +242,32 @@ void AddResult(SearchResult result)
 				}
 				dispatcher.BeginInvoke(
 					DispatcherPriority.Normal,
-					new Action(delegate { this.Results.Insert(this.Results.Count - 1, result); }));
+					new Action(delegate { InsertResult(result); }));
 				cts.Token.ThrowIfCancellationRequested();
 			}
 
+			void InsertResult(SearchResult result)
+			{
+				if (Options.DisplaySettingsPanel.CurrentDisplaySettings.SortResults)
+				{
+					// Keep results collection sorted by "Fitness" by inserting result into correct place
+					// Inserts in the beginning shifts all elements, but there can be no more than 1000 items.
+					for (int i = 0; i < this.Results.Count; i++)
+					{
+						if (this.Results[i].Fitness < result.Fitness)
```

Comment:
What happens if the fitness is not set properly? The result is never inserted. The old code always inserted at the end, maybe you should do the same as a fallback